url529c09a30c_kianyangchn_DriveNow_tgz-pJ8-drivenow_tests_TestJ8/src/drivenow/controls/Unterrichtsverwaltung.java

```diff
@@ -0,0 +1,309 @@
+package drivenow.controls;
+
+import java.util.ArrayList;
+import java.util.Date;
+
+import drivenow.Systemereignis;
+import drivenow.Systemereignis.Nachricht;
+import drivenow.entities.*;
+import drivenow.interfaces.ISekretaerBoundary.Stundenart;
+
+public class Unterrichtsverwaltung {
+
+	private static Unterrichtsverwaltung instance;
+	private ArrayList<Theoriestunde> theoriestunde;
+	private ArrayList<Fahrstunde> fahrstunde;
+	private Ressourcenverwaltung ressourcenverwaltung;
+	
+	public Unterrichtsverwaltung() {
+		this.theoriestunde = new ArrayList<Theoriestunde>();
+		this.fahrstunde = new ArrayList<Fahrstunde>();
+		this.ressourcenverwaltung = Ressourcenverwaltung.getInstance();
+	}
+	
+	public static Unterrichtsverwaltung getInstance() {
+		if (instance == null) {
+			instance = new Unterrichtsverwaltung();
+		}
+		return instance;
+	}
+	
+	public void cleanInstance() {
+		instance=null;
+	}
+	
+	private Fahrschueler find_fahrschueler(String schuelerID) {
+		ArrayList<Fahrschueler> fahrschueler = this.ressourcenverwaltung.get_fahrschueler();
+		if (fahrschueler.size()!=0) {
+			for (int i=0;i<fahrschueler.size();i++) {
+				if (fahrschueler.get(i).get_ID()==schuelerID) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.